Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/SaveNewCampaignLinks.cs

```diff
@@ -0,0 +1,52 @@
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Doppler.HtmlEditorApi.Storage.DapperProvider.Queries;
+
+/// <summary>
+/// It keeps existing DB entries and only adds new ones without deleting or updating anything.
+/// </summary>
+public record SaveNewCampaignLinks(
+    int IdContent,
+    IEnumerable<string> Links
+) : IExecutableDbQuery
+{
+    const string BASE_QUERY = @"
+INSERT INTO [Link] (IdCampaign, UrlLink, IsActiveForTracking, IsDynamic)
+SELECT rest.IdCampaign, newLinks.UrlLink, rest.IsActiveForTracking, rest.IsDynamic
+FROM
+    (SELECT @IdCampaign AS IdCampaign, 1 AS IsActiveForTracking, 0 AS IsDynamic) rest
+JOIN (
+    {{linksSelect}}
+    ) AS newLinks ON 1=1
+LEFT JOIN [Link] oldLinks ON oldLinks.UrlLink = newLinks.UrlLink AND oldLinks.IdCampaign = rest.IdCampaign
+WHERE oldLinks.IdCampaign IS NULL";
+
+    // The result is something like:
+    //    SELECT @Url1 AS UrlLink
+    //    UNION SELECT @Url2 AS UrlLink
+    //    UNION SELECT @Url3 AS UrlLink
+    public string GenerateSqlQuery()
+        => BASE_QUERY.Replace(
+            "{{linksSelect}}",
+            string.Join(
+                "\n    UNION ",
+                Enumerable.Range(1, Links.Count())
+                .Select(x => $"SELECT @Url{x} AS UrlLink")));
```

Comment:
@MarianoHerrera could you review this query?

Reply:
https://user-images.githubusercontent.com/6794856/160184480-584887a9-4524-467d-ac6e-f69ea033de65.mp4